lib/src/components/NxDropdownIconOnly/__tests__/NxDropdownIconOnly.test.tsx

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import * as enzymeUtils from '../../../__testutils__/enzymeUtils';
+import { faCog } from '@fortawesome/free-solid-svg-icons';
+
+import NxDropdownIconOnly, { Props } from '../NxDropdownIconOnly';
+import NxButton from '../../NxButton/NxButton';
+import NxOverflowTooltip from '../../NxTooltip/NxOverflowTooltip';
+import NxDropdownMenu from '../../NxDropdownMenu/NxDropdownMenu';
+import NxFontAwesomeIcon from '../../NxFontAwesomeIcon/NxFontAwesomeIcon';
+
+describe('NxDropdownIconOnly', () => {
+  let container: HTMLDivElement | null;
+
+  const minimalProps = {
+        title: 'Test tooltip',
+        isOpen: false,
+        onToggleCollapse: () => {}
+      },
+      getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxDropdownIconOnly, minimalProps),
+      getMountedComponent = enzymeUtils.getMountedComponent<Props>(NxDropdownIconOnly, minimalProps);
+
+  beforeEach(function() {
+    // Avoid rendering directly on the body.
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(function() {
+    if (container) {
+      document.body.removeChild(container);
+      container = null;
+    }
+  });
+
+  it('renders a button with the appropriate classes and type=button', function() {
+    const component = getShallowComponent(),
+        button = component.find(NxButton);
+
+    expect(component).toHaveClassName('.nx-dropdown--icon-only');
+    expect(button).toHaveClassName('.nx-dropdown__toggle');
+    expect(button).toHaveProp('variant', 'icon-only');
+    expect(button).toHaveProp('type', 'button');
+  });
+
+  it('renders the button according to the supplied variant', function() {
+    let component = getShallowComponent();
+    expect(component.find(NxButton)).toHaveProp('variant', 'icon-only');
+  });
+
+  it('correctly renders a custom icon based on icon prop', function() {
```

Comment:
Is there a test of the default icon?

Reply:
> Is there a test of the default icon?

There is now: https://github.com/sonatype/sonatype-react-shared-components/pull/465/commits/107379d31f976dd19db131e2e8fd1cd5da73f14b